gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSourceWeekly.java

```diff
@@ -0,0 +1,51 @@
+package gobblin.ingestion.google.webmaster;
+
+import gobblin.configuration.WorkUnitState;
+import gobblin.source.extractor.extract.LongWatermark;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.Pair;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+
+public class GoogleWebMasterSourceWeekly extends GoogleWebMasterSource {
+  private static DateTimeFormatter _hmsFormatter = DateTimeFormat.forPattern("yyyyMMddHHmmss");
+
+  @Override
+  GoogleWebmasterExtractor createExtractor(WorkUnitState state, Map<String, Integer> columnPositionMap,
+      List<GoogleWebmasterFilter.Dimension> requestedDimensions,
+      List<GoogleWebmasterDataFetcher.Metric> requestedMetrics) throws IOException {
+
+    long lowWatermark = state.getWorkunit().getLowWatermark(LongWatermark.class).getValue();
```

Comment:
Under what condition will this be null? QueryBasedSource guarantees workunit to have low water mark at line 130. I thought only work units with water marks could be created. If it would have a null exception, what could I do? Wrap it in an IOException? Which IOException should I use?